### ISSUE TITLE
[Bug]: Crash at com.onesignal.common.modeling.Model.toJSON (Model.kt:312) 

### ISSUE THREAD

**Issue by jvojacek** [OneSignal/OneSignal-Android-SDK#2040]
### What happened?

Fatal Exception: java.util.ConcurrentModificationException:
       at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:760)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:792)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:790)
       at com.onesignal.common.modeling.Model.toJSON(Model.kt:312)
       at com.onesignal.common.modeling.ModelStore.persist(ModelStore.kt:141)
       at com.onesignal.common.modeling.SingletonModelStore.replace(SingletonModelStore.kt:36)
       at com.onesignal.core.internal.config.impl.ConfigModelStoreListener$fetchParams$1.invokeSuspend(ConfigModelStoreListener.kt:101)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:32)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)

### Steps to reproduce?

```Markdown
This crash is reported by crashlytics
```

### What did you expect to happen?

App should not crash

### OneSignal Android SDK version

5.0.2

### Android version

13

### Specific Android models

```Markdown
Galaxy A53 5G on Android 14
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by michael-winkler** [comment]
You should update to the latest version of OneSignal `5.1.7`

**Comment by jkasten2** [comment]
@jvojacek I can confirm this was fixed in [5.1.1](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.1), but we always recommend the latest version to ensure you have the latest bug fixes and features available. 

Duplicate of issue #1856
